src/realm/sync/network/default_socket.cpp

```diff
@@ -7,24 +7,28 @@
 namespace realm::sync::websocket {
 
 namespace {
-class EZSocketImpl final : public EZSocket, public Config {
+class DefaultWebSocketImpl final : public WebSocketInterface, public Config {
 public:
-    EZSocketImpl(EZConfig& config, EZObserver& observer, EZEndpoint&& endpoint)
-        : m_logger_ptr{config.logger}
+    DefaultWebSocketImpl(const std::shared_ptr<util::Logger>& logger_ptr, network::Service& service,
+                         std::mt19937_64& random, const std::string user_agent, EZObserver& observer,
+                         EZEndpoint&& endpoint)
+        : m_logger_ptr{logger_ptr}
         , m_logger{*m_logger_ptr}
-        , m_random{config.random}
-        , m_service{config.service}
-        , m_user_agent{config.user_agent}
+        , m_random{random}
+        , m_service{service}
+        , m_user_agent{user_agent}
         , m_observer{observer}
         , m_endpoint{std::move(endpoint)}
         , m_websocket(*this)
     {
         initiate_resolve();
     }
 
-    void async_write_binary(const char* data, size_t size, util::UniqueFunction<void()>&& handler) override
+    void async_write_binary(util::Span<const char> data, SyncSocketProvider::FunctionHandler&& handler) override
```

Comment:
I think the previous signature will work better with coming the c-api

Reply:
The span is just a wrapper to the data pointer and size - you can create it using `util::Span<const char>{data, size}` - and it can be used similar to a std::container (iterators, index operator, etc.) with bounds checking.